RoT/Protocol/RFCs/0001-Transport_Agnosticism.md

```diff
@@ -0,0 +1,149 @@
+* Name: Transport_Agnosticism
+* Date: 2021-04-02
+* Pull Request: [#16](https://github.com/opencomputeproject/Security/pull/16)
+
+# Objective
+
+Cerberus currently mandates that its messages be exchanged over an MCTP bus,
+using a Microsoft-specific PCI vendor ID. This is a limitation for depoyments
+that might want to speak Cerberus over a completely different bus, such as
+SPI, I3C, exotic busses like NVME, or over plain old TCP or UDP for the
+purposes of conformance testing.
+
+This RFC describes a "transport agnostic" model for Cerberus with the following
+goals:
+- The existing MCTP binding of the protocol works with minimal (or no) changes.
+- Cerberus can be used over an arbitrary transport layer without directly
+  implementing special support for it in a generic Cerberus library.
+- References to MCTP in the Cerberus Challenge Protocol specification will be
+  moved to an appendix, describing it as a possible option for transporting
+  Cerberus.
+
+# Proposal
+
+"Transport agnostic" Cerberus specifies an *admissible transport* as any protocol
+or bus that satisfies a set of properties. These properties reflect the
+properties of MCTP already used by Cerberus.
+
+1. An admissible transport is a mechanism for sending a *message* (a dynamically-\
+   sized byffer of bytes) from one (not necesarily addressable) endpoint to
+   another. In other words, the admissible transport is responsible for removing
+   frames from packets and assembling them in sequence.
+
+2. Each endpoint has a known *maximum message length*, measured in bytes. The
+   transport is responsible for negotiating this parameter for each device a
+   Cerberus device intends to speak to. This parameter must be made available
+   to the Cerberus protocol but Cerberus itself does not negotiate it. This
+   paramter must be at least 64 bytes.
+
+3. Each message is either the request or response half of a Cerberus command, as
+   defined in the Challenge Protocol. To uniquely identify the type of an
+   incoming message, the binding of Cerberus to the admissible transport must
+   specify a *transport-specific header* that contains at a minimum the folling
+   parameters. How they are encoded is irrelevant to Cerberus, beyond that they
+   be computable without recieving the entire incoming message:
+   - The command type byte (as specified in the Challenge Protocol).
+   - A single bit: whther this message is the request or response half of the
```

Comment:
Is it necessary to call out 'a single bit' here?  Seems to contradict the previous statement of encoding being irrelevant.

Reply:
I think "a boolean" is probably more appropriate?

Reply:
Boolean probably works, but it feels like you don't even need to be that specific.  Simply stating that the transport should understand whether the message is request or response seems sufficient.  Completely stays out of implementation details.